js/web/lib/wasm/jsep/webgpu/ops/resize.ts

```diff
@@ -0,0 +1,347 @@
+          ((opsetVersion < 11) ? true :
+                                 (attributes.mode === Mode.nearest &&
+                                  attributes.coordinateTransformMode === CoordinateTransformMode.asymmetric &&
+                                  attributes.nearestMode === NearestMode.floor)) &&
+          scales.toString() === '1,1,2,2';
+      const useNearest2xOptimizationSnippet = `
+        inputIndices[0] = outputIndices[0]; // batch size
+        inputIndices[1] = outputIndices[1]; // channel
+        inputIndices[2] = outputIndices[2] / 2; // height
+        inputIndices[3] = outputIndices[3] / 2; // width
+      `;
+      const getShaderSource = (shaderHelper: ShaderHelper) => `
+      ${attributes.mode === Mode.nearest ? getNearstPixelFromOriginal(attributes.nearestMode) : ';'};
+      ${getOriginalCoordinateFromResizedCoordinate(attributes.coordinateTransformMode)};
+      @group(0) @binding(0) var<storage, read> input : array<${dataType}>;
+      @group(0) @binding(1) var<storage, read_write> output : array<${dataType}>;
+      ${outputIndicesHelper.o2iImpl}
+      ${inputIndicesHelper.i2oImpl}
+      ${shaderHelper.mainStart()}
+        ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes(outputSize)}
+        if (${noScale}) {
+          output[global_idx] = input[global_idx];
+        }
+        else
+        {
+          ${outputIndicesHelper.indicesVariableDeclaration('outputIndices')}
+          ${outputIndicesHelper.o2iCall('global_idx', 'outputIndices')}
+          ${inputIndicesHelper.indicesVariableDeclaration('inputIndices')}
+          if (${useNearest2xOptimization}) {
+            ${useNearest2xOptimizationSnippet}
+          } else {
+            var value = ${dataType}(0);
+            // TODO: calculate mapping input indices
+            value += input[${inputIndicesHelper.i2oExpression('inputIndices')}];
+            output[global_idx] = value;
+          }
+        }
+      }`;
+
+      return {
+        ...metadata,
+        getShaderSource,
+        outputs: [{dims: outputShape, dataType: inputs[0].dataType, gpuDataType: GpuDataType.default}],
+        dispatchGroup: () => ({x: Math.ceil(outputSize / 64 /* workgroup size */)})
+      };
+    };
+
+export const createResizeProgramInfoLoader =
+    (inputs: readonly TensorView[], attributes: ResizeAttributes): ProgramInfoLoader => {
+      const metadata: ProgramMetadata = {
+        name: 'Resize',
+        inputTypes: [GpuDataType.default],
+        cacheHint: attributes.cacheKey,
+      };
+      return {...metadata, get: () => createResizeProgramInfo(metadata, inputs, attributes)};
+    };
+
+const readCustomDataBuffer = (context: ComputeContext): void => {
+  const customDataBuffer = context.customDataBuffer;
+  const customDataBuffer32 = new Uint32Array(customDataBuffer, customDataBuffer.byteOffset, 1);
+  opsetVersion = customDataBuffer32[0];
+};
+
+export const resize = (context: ComputeContext, attributes: ResizeAttributes): void => {
```

Comment:
so this is the entry.

you may need to define a type, say `ResizeContext`, in typescript, like this:
```ts
interface ResizeContext {
    readonly computeContext: ComputeContext; // store computeContext here
    readonly opset: number; 
    // and other things
}
```

`readCustomDataBuffer()` return an instance of this type (maybe rename the function).

then modify functions to accept ResizeContext as parameter. the per-call information will be carried by this object.

Reply:
Renamed the function as getOpsetVersionFromCustomDataBuffer. The opsetVersion is a local variable or a function parameter, not a global variable. May be a new abstraction/wrapper ResizeContext is not necessary.

Reply:
sure. if `opset` is the only data being carried, no need to create a wrapper like ResizeContext for that.